src/storage/mod.rs

```diff
@@ -1163,41 +1163,56 @@ mod tests {
         assert_eq!(x.unwrap().unwrap(), v);
     }
 
+    fn expect_scan(v: Vec<Option<KvPair>>, x: Result<Vec<Result<KvPair>>>) {
+        let x: Vec<Option<KvPair>> = x.unwrap().into_iter().map(Result::ok).collect();
+        assert_eq!(x, v);
+    }
+
+    fn expect_batch_get_vals(v: Vec<Option<KvPair>>, x: Result<Vec<Result<KvPair>>>) {
+        let x: Vec<Option<KvPair>> = x.unwrap().into_iter().map(Result::ok).collect();
+        assert_eq!(x, v);
+    }
+
+    fn expect_result_error<T, F>(err_matcher: F, x: Result<T>)
```

Comment:
It seems more intuitive to put the result before the error matcher.

Reply:
Currently these expect_xxx 's signature is like (expected, got). Should I change them all?

Reply:
I don't know if it is personal, if you think it is fine, you can keep it.